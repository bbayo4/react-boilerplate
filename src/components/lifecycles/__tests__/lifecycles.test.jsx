import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import Lifecycles from '../lifecycles';

describe('The Lifecyles Component', () => {

  const wrapper = shallow(<Lifecycles />);
  const componentPrototype = Lifecycles.prototype;

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('calls componentWillUpdate after toggle box button is clicked', () => {
    jest.spyOn(componentPrototype, 'componentWillUpdate');
    const toggleBoxBtn = wrapper.find('button').at(0);
    toggleBoxBtn.simulate('click');
    expect(componentPrototype.componentWillUpdate.mock.calls.length).toBe(1);
  });

  test('calls componentWillUnmount when necessary', () => {
    jest.spyOn(componentPrototype, 'componentWillUnmount');
    wrapper.unmount();
    expect(componentPrototype.componentWillUnmount).toHaveBeenCalled();
  });

});