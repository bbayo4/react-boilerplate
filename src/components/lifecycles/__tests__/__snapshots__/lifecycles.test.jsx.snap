// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Lifecyles Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Lifecycles />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          BASIC LIFECYCLE METHODS...
        </h1>,
        false,
        <button
          onClick={[Function]}
        >
          toggle box
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BASIC LIFECYCLE METHODS...",
        },
        "ref": null,
        "rendered": "BASIC LIFECYCLE METHODS...",
        "type": "h1",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "toggle box",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "toggle box",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            BASIC LIFECYCLE METHODS...
          </h1>,
          false,
          <button
            onClick={[Function]}
          >
            toggle box
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BASIC LIFECYCLE METHODS...",
          },
          "ref": null,
          "rendered": "BASIC LIFECYCLE METHODS...",
          "type": "h1",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "toggle box",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "toggle box",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
